### DISCUSSION TITLE
Command to disable/enable motion detection

### DISCUSSION THREAD

**Discussion by BillyCroan** [tsightler/ring-mqtt#753] [General Discussion]
is there a command with ring-mqtt to disable and enable motion detection on ringcam or ring doorbell?  I'd like to script it cams to disable motion on mowing day and when our doors are open to reduce unnecessary alerts.

**Comment by tsightler** [comment]
Yes, I added the topics to the wiki documentation and the switch should appear in Home Assistant if using it.

**Reply by BillyCroan** [reply]
Thanks, based on the docs now, it looks like the command look like this:
`mosquitto_pub -t ring/9yutvj-4ive-0/camera/5c475e625347/motion_detection/command -m OFF`
to disable motion detection and ON to enable it.

It seems to work for me, but of note, the ring app does not shoe the slider move when changing this setting.  instead one must exit the settings screen and go back in to witness the change.

In the interest of reducing latency and complexity, is there a way to send these commands to the ring api without using an MQTT server?  There isn't much time between opening the door, and a motion event from walking through it.  I need the setting to be applied in the second and a half between opening the door, and walking through it.  might only be 1200ms

**Reply by tsightler** [reply]
> It seems to work for me, but of note, the ring app does not shoe the slider move when changing this setting. instead one must exit the settings screen and go back in to witness the change.

No reason it wouldn't work for you just like it does for anybody else!  :)  Ring app never updates in real-time, this is no different if you have two devices with the Ring app open, changes in one app aren't reflected in the other app immediately, usually require a manual refresh (exit/re-enter) although some screen have a slow polling refresh (60 seconds or so).  That's why the ring-client-api has a polling interval, because it's a polled API.

> In the interest of reducing latency and complexity, is there a way to send these commands to the ring api without using an MQTT server? 

It's not really clear to me why using MQTT isn't already considered simple.  MQTT was built for sending low-latency messages between machines using minimal resources.  MQTT itself certainly won't add any significant latency at all.  The entire reason I built ring-mqtt is because I wanted near real-time status from Ring Alarm sensors, and this gets very, very close (sub-100ms).  Commands rarely take more than 200ms, and 95% of that time is the Ring REST API, not MQTT or anything in ring-mqtt itself.  I don't think there's any way that running a new node process each time wouldn't be much heavier.

The only way I could think of that might make it faster was to send the request directly to the Ring API using curl or something, but you'll need a token do to this and then you'll need to refresh that token every so often or it will expire.  Using ring-client-api does all of that for you.

**Reply by BillyCroan** [reply]
> Using ring-client-api does all of that for you.

I think that is the path i want to go down.  But pardon me I'm new to this project.  ring-mqtt and ring-client-api are two different things right?  I can see how to kick off a MQTT command from a commandline.  But I don't see how to use ring-client-api from cli.

My door position detector is a python script that checks a hall-effect sensor on an rpi every 5ms.  from that, I can kick off an http post or cli execution fairly easily.  Maintaining the token freshness is worth keeping something running and if you have that figured out I'd rather not reinvent THAT wheel.

ring-client-api is looking to be just a library that has to be run by another program.  I assume ring-mqtt does that and its ring-mqtt that keeps the token alive not ring-client-api, right?

**Reply by tsightler** [reply]
ring-mqtt uses ring-client-api, there's no CLI.  Becuase ring-client-api is written in Typescript, it needs a server runtime, typically NodeJS.  You could write a script that uses ring-client-api and runs under NodeJS, but firing up a NodeJS process and loading ring-client-api just to send a single command is very heavy and will take 100's of ms at least.  Ring-mqtt runs constantly, maintaining the connection, so you only need to send a command and it's immediately forwarded. 

To me it makes zero sense to manually try to reproduce that when ring-mqtt is already there allowing you to access what you need via a standard protocol (MQTT).  If you wanted you could just modify your python script to send the MQTT command directly, saving having to spawn mosquitto_pub.  But you can do whatever you want.

**Reply by tsightler** [reply]
BTW, since you are using Python, maybe you'd be better off to just use the [Ring Python API](https://github.com/tchellomello/python-ring-doorbell).  You could integrate that directly into your existing Python script to call the API although I admittedly don't know if it has the enable/disable motion function.  Technically ring-client-api doesn't have this either, I just manually construct the required REST call in ring-mqtt and send it directly usign the ring-client-api restClient functions.

**Reply by tsightler** [reply]
Looking at recent updates to [Ring Python API](https://github.com/tchellomello/python-ring-doorbell) it looks like they've introduced a CLI and it even has an enable/disable motion detect option, so you can probably just use that directly from your Python script.
